### PR TITLE
[Silabs] Add Silabs Platform abstraction

### PR DESCRIPTION
**Pull request by jepenven-silabs**
Add Abstraction of SDK used in our examples

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

File: src/platform/silabs/platformAbstraction/WiseMCU_SPAM.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+CHIP_ERROR SilabsPlatform::Init(void)
+{
+    init_ccpPlatform();
+}
+
+#ifdef ENABLE_WSTK_LEDS
+void SilabsPlatform::InitLed(void)
+{
+    // TODO ?
```
Comment (andy31415) [inline src/platform/silabs/platformAbstraction/WiseMCU_SPAM.cpp:40]: This todo is confusing...should it add details? same below for GetLedState

File: src/platform/silabs/platformAbstraction/WiseMCU_SPAM.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+bool SilabsPlatform::GetLedState(uint8_t led)
+{
+    // TODO ?
+    return SilabsPlatformAbstractionBase::GetLedState(led);
+}
+
+CHIP_ERROR SilabsPlatform::ToggleLed(uint8_t led) override
+{
+    // TODO add range check ?
+    RSI_Board_LED_Toggle(led) return CHIP_NO_ERROR;
```
Comment (andy31415) [inline src/platform/silabs/platformAbstraction/WiseMCU_SPAM.cpp:60]: is there a `;` missing here?

**Review by andy31415 — APPROVED** [review]

File: src/platform/silabs/platformAbstraction/GSDK_SPAM.cpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    (state) ? sl_led_turn_on(SL_SIMPLE_LED_INSTANCE(led)) : sl_led_turn_off(SL_SIMPLE_LED_INSTANCE(led));
+    return CHIP_NO_ERROR;
+}
+
+bool SilabsPlatform::GetLedState(uint8_t led)
+{
+    if (led >= SL_SIMPLE_LED_COUNT)
+    {
+        return 0;
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/platform/silabs/platformAbstraction/GSDK_SPAM.cpp:57]: will do in the follow up PR

File: src/platform/silabs/platformAbstraction/WiseMCU_SPAM.cpp (hunk 2 above)
Comment (jepenven-silabs) [reply src/platform/silabs/platformAbstraction/WiseMCU_SPAM.cpp:60]: will do in the follow up PR

File: src/platform/silabs/platformAbstraction/GSDK_SPAM.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/platform/silabs/platformAbstraction/GSDK_SPAM.cpp:57]: done

[the remark this answers is not in the document]
Comment (andy31415) [reply src/platform/silabs/platformAbstraction/GSDK_SPAM.cpp:57]: Diff does not show it as done....

**Pull request merged**